resources/measures/upgrade_hvac_condensing_boiler/measure.rb

```diff
@@ -0,0 +1,251 @@
+
+    sizing_systems = model.getSizingSystems
+    sizing_systems.each do |sizing_system|
+      # sizing_system.autosizeHeatingDesignCapacity
+    end
+
+    # set empty array for hot water coils
+
+    boilers = model.getBoilerHotWaters
+    boilers.each do |boiler|
+      # get existing fuel type and efficiency
+      existing_boiler_fuel_type = boiler.fuelType
+      existing_boiler_efficiency = boiler.nominalThermalEfficiency
+      existing_boiler_capacity = boiler.nominalCapacity
+      existing_boiler_name = boiler.name
+
+      runner.registerInfo("Existing boiler #{existing_boiler_name} has nominal efficiency #{existing_boiler_efficiency}, fuel type #{existing_boiler_fuel_type}, and nominal capacity #{existing_boiler_capacity} W.")
+
+      # set efficiency to 0.95 to reflect condensing boiler
+      boiler.setNominalThermalEfficiency(0.95)
+
+      # set new performance curve
+      boiler.setNormalizedBoilerEfficiencyCurve(condensing_boiler_curve)
+      boiler.setEfficiencyCurveTemperatureEvaluationVariable('EnteringBoiler')
+      boiler.setWaterOutletUpperTemperatureLimit(hw_setpoint_c)
+
+      # autosize the boiler capacity. should not make a difference if running standalone measure, but could downsize if run with other upgrades.
+      # boiler.autosizeDesignWaterFlowRate
+      # boiler.autosizeNominalCapacity
+
+      if existing_boiler_name.get.include?('Supplemental')
+        # denotes boiler is a supplemental boiler located on a condenser loop. do not resize pump or reset loop temp.
+        boiler.setName('Supplemental Condensing Boiler Thermal Eff 0.95')
+      else
+        # get plant loop and set new supply temp and setpoint manager to 140F
+        boiler.inletModelObject.get.to_Node.get
+        boiler.outletModelObject.get.to_Node.get
+        htg_loop = boiler.plantLoop.get
+        htg_loop_sizing = htg_loop.sizingPlant
+        htg_loop_sizing.setDesignLoopExitTemperature(hw_setpoint_c)
+        htg_loop_sizing.setLoopDesignTemperatureDifference(hw_delta_t_c)
+        # htg_loop.autosizeMaximumLoopFlowRate
+        # htg_loop.autocalculatePlantLoopVolume
+
+        htg_loop.supplyOutletNode.setpointManagers.each do |spm|
+          spm.remove
+        end
+
+        # original method: set sp manager to constant 140F
+        # spm.to_SetpointManagerScheduled.get.setSchedule(new_setpoint_sched)
+
+        # create new OA reset setpoint manager with settings: hw temp 140f for air temp <20F, hw temp 120F for air temp >50F
+        oa_reset_sp_mgr = OpenStudio::Model::SetpointManagerOutdoorAirReset.new(model)
+        oa_reset_sp_mgr.setSetpointatOutdoorHighTemperature(OpenStudio.convert(140, 'F', 'C').get)
+        oa_reset_sp_mgr.setOutdoorHighTemperature(OpenStudio.convert(50, 'F', 'C').get)
+        oa_reset_sp_mgr.setSetpointatOutdoorLowTemperature(OpenStudio.convert(180, 'F', 'C').get)
+        oa_reset_sp_mgr.setOutdoorLowTemperature(OpenStudio.convert(20, 'F', 'C').get)
```

Comment:
Nice idea. But the comment  (which is ok) don't match the values (they are wrong). Also, reuse the hw_setpoint_f to avoid falling out of sync.